### ISSUE TITLE
Alarms: not being cleared when IOC is stopped

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#3335]
As a user on SANDALS, I do not want to be worried because the GUI is displaying alarms for IOCs which are running

Find out (& fix) why alarms sometimes stay behind after an IOC is stopped.

**Notes**
- Reported by SANDALS, their alarm server was reporting eurotherm errors, however no eurotherms were running.

**Comment by davidkeymer** [comment]
IOC message relating to problem:
`[1531495389.81] MAJOR: Could not restart IOC ALARM: Could not get auto-restart property for IOC (TE:NDW1890:CS:PS:ALARM) got '¨`

**Comment by Tom-Willemsen** [comment]
I wasn't able to reproduce this issue consistently. The error above indicates that genie_python's get_pv layer returned something that wasn't expected. I found a threading condition which may have caused this, which is fixed in https://github.com/ISISComputingGroup/genie_python/pull/147 .

I've also made the blockserver retry getting the value if it was invalid in https://github.com/ISISComputingGroup/EPICS-inst_servers/pull/175/files

I think we should review the two changes above and if we still get the issue in a later release (with these changes) then re-open a new ticket for it.
